### DISCUSSION TITLE
how to add new modules?

### DISCUSSION THREAD

**Discussion by Hanmeim** [mu-editor/mu#2307] [Q&A] [answered]
I would like to teach my students to use numpy & matplotlib to modify some pictures. But it seems mu is not using the "python" directory in its root as its lib. Instead it uses a newly created "venv bla bla" directory. so how to add new modules into it? I've tried pip but only end up in old python dir and just copying into venv dir resulted in problematic coding experience which might be a result of unmatching lib version. 
So, how to add new modules to mu?

**Comment by carlosperate** [comment, marked as answer]
Hi @Hanmeim,

You can find this info in the Mu website: https://codewith.mu/en/tutorials/1.1/pypi
